Tests/FluentAssertions.Equivalency.Specs/DictionarySpecs.cs

```diff
@@ -1188,4 +1189,170 @@ public void Passing_the_reason_to_the_inner_equivalency_assertion_works()
 
         act.Should().Throw<XunitException>().WithMessage("*FOO BAR*");
     }
+
+    [Fact]
+    public void A_non_generic_subject_can_be_compared_with_a_generic_expectation()
+    {
+        var subject = new ListDictionary
+        {
+            ["id"] = 22,
+            ["CustomerId"] = 33
+        };
+
+        var expected = new Dictionary<object, object>
+        {
+            ["id"] = 22,
+            ["CustomerId"] = 33
+        };
+
+        subject.Should().BeEquivalentTo(expected);
+    }
+
+    [Fact]
+    public void A_non_generic_subject_which_is_null_can_be_compared_with_a_generic_expectation()
+    {
+        var subject = (ListDictionary)null;
+
+        var expected = new Dictionary<object, object>
+        {
+            ["id"] = 22,
+            ["CustomerId"] = 33
+        };
+
+        Action act = () => subject.Should().BeEquivalentTo(expected);
+
+        act.Should().Throw<XunitException>().WithMessage("*<null>*");
+    }
+
+    [Fact]
+    public void Excluding_nested_objects_when_dictionary_is_equivalent()
+    {
+        var subject = new Dictionary<object, object>
+        {
+            ["id"] = 22,
+            ["CustomerId"] = 33
+        };
+
+        var expected = new Dictionary<object, object>
+        {
+            ["id"] = 22,
+            ["CustomerId"] = 33
+        };
+
+        subject.Should().BeEquivalentTo(expected, opt => opt.ExcludingNestedObjects());
+    }
+
+    [Fact]
+    public void Custom_types_which_implementing_dictionaries_pass()
+    {
+        var subject = new MyDictionary1();
+        subject.Add("id", 22);
+        subject.Add("CustomerId", 33);
+
+        var expected = new MyDictionary2
+        {
+            { "id", 22 },
+            { "CustomerId", 33 }
+        };
+
+        subject.Should().BeEquivalentTo(expected);
+    }
+
+    [Fact]
+    public void Custom_types_which_implementing_dictionaries_pass_with_swapped_subject_expectation()
+    {
+        var subject = new MyDictionary2
+        {
+            { "id", 22 },
+            { "CustomerId", 33 }
+        };
+
+        var expected = new MyDictionary1();
+        expected.Add("id", 22);
+        expected.Add("CustomerId", 33);
+
+        subject.Should().BeEquivalentTo(expected);
+    }
+}
+
+internal class MyDictionary1 : Dictionary<string, int>
+{
+    public new void Add(string key, int value)
+    {
+        base.Add(key, value);
+    }
+}
+
+internal class MyDictionary2 : IDictionary<string, int>
```

Comment:
I just tried with your example above :)

I will change them.

Are your suggestions meant as "either one" or `MyDictionary1 `-> `NonGenericChildDictionary`; `MyDictionary2 `-> `NonGenericDictionary`

Reply:
As a side-note: Your example above was `class MyDictionary2: IDictionary<string, int>, IDictionary`, but this does not work, because either one interface requires a property `Keys` and a property `Values`, which cannot live together in one class.

The same with a few other methods